src/cli/notificationsProcessQueue.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+	/**
+  * @version    SVN: <svn_id>
+  * @package    PIP
+  * @copyright  Copyright (C) 2005 - 2014. All rights reserved.
+  * @license    GNU General Public License version 2 or later; see LICENSE.txt
+  * This script updates the masterlist terms into solr autosuggest core
+  * Based on the config set.
+  * is derivative of works licensed under the GNU General Public License or
+  * other free or open source software licenses.
+  */
+
+define('_JEXEC', 1);
+define('JPATH_BASE', realpath(dirname(__FILE__) . '/..'));
+
+require_once JPATH_BASE . '/includes/defines.php';
+require_once JPATH_BASE . '/includes/framework.php';
+require_once JPATH_BASE . '/libraries/import.php';
+require_once JPATH_CONFIGURATION . '/configuration.php';
+
+jimport('joomla.application.cli');
+
+/*
+error_reporting(0);
+ini_set('display_errors', 'On');
+*/
+
+jimport('joomla.filesystem.file');
+
+if (!JComponentHelper::getComponent('com_tjreports', true)->enabled)
+{
+	return;
+}
+
+jimport('joomla.log.log');
+
+/**
+ * This is used to send dequeue TJ notifications. Currently, we've written snippet for emails which can be extented
+ * to support other notifications.
+ *
+ * @since  1.0.0
+ */
+class ProcessNotifications extends JApplicationCli
+{
+	/**
+	 * This gets data from reminder email table & sent emails.
+	 *
+	 * @return  Nothing
+	 *
+	 * @since   1.0.0
+	 */
+	public function execute()
+	{
+		$db			=	JFactory::getDBO();
+		$mainframe	=	JFactory::getApplication('site');
+		$query		=	$db->getQuery(true);
+		$limit		=	$this->input->get('limit', 20, 'INT');
+
+		$query->select('*');
+		$query->from('#__tj_notification_queue');
+		$query->where("status = 0");
+		$query->setLimit($limit);
+		$db->setQuery($query);
+		$queueData	=	$db->loadObjectList();
+
+		if (count($queueData) <= 0)
+		{
+			return $this->out('Notification queue is empty!');
+		}
+
+		$from		=	$mainframe->getCfg('mailfrom');
+		$fromname	=	$mainframe->getCfg('fromname');
+		$sentCount	=	0;
+
+		foreach ($queueData as $data)
+		{
+			$options	= json_decode($data->options, true);
+			$mailer		= JFactory::getMailer();
+
+			if ($options['emailfrom'] != null && $options['emailfromname'] != null)
```

Comment:
You're assuming that only emails will go in queue. What about sms, push and other types ?